### ISSUE TITLE
No way to wait for all failed activations

### ISSUE THREAD

**Issue by LachlanStuart** [pywren/pywren-ibm-cloud#254]
When debugging problems inside activations, it can be helpful to get a list of all failed activations so that you can see how common the problem is, and if they all hit the same problem or have different problems. However, at the moment PyWren raises an exception once it hits the first failure, and doesn't seem to provide any option to avoid this behavior.

Example:
```
def cycle_colors(i):
    options = ['red', 'green', 'blue']
    return options[i % 4]
        
pw = pywren.ibm_cf_executor(workers=2)
pw.map(cycle_colors, range(10))
colors = pw.get_result()
```

The current behavior of stopping on the first error is good for production use, but ideally there would be some parameter to wait for all errors/results.

There seem to be two underlying problems:
1. The executor always stops launching new activations and cleans up when it detects a failure. This means that many activations don't start, and several that are in the process of starting can fail because their job data has been cleaned up.
2. `wait()` and `get_result()` don't seem to have any option to keep running after they've found an exception. I expected `throw_except` to control this, but that seems to only switch between how the first exception is handled. It doesn't actually seem to continue waiting after it finds that exception.

This issue also means that it's currently impossible to cleanly implement "retry with more resources" behavior. E.g. if you have a small job with unpredictable memory usage, you may want to run with `runtime_memory=512`, and re-run any failed jobs with `runtime_memory=2048` or with smaller inputs per job.

For context, I'm using the ibm_cf_executor without RabbitMQ. I'm not sure if the RabbitMQ behavior is different.

**Comment by JosepSampe** [comment]
@LachlanStuart `throw_except` param behavior is already fixed in master branch. I think #255 is also required. So by setting `throw_except=False`, the invoker will never stop, and you can wait for all failed activations and mesure how common is the errror.

**Comment by LachlanStuart** [comment]
@JosepSampe It's awesome that you're one step ahead of me! However, I get an exception using `throw_except=False` in 1.4.0 and also the #255 branch:

```
Traceback (most recent call last):
  File "./replicate_timeout_issue_with_creds.py", line 161, in <module>
    pw.get_result(all_futures, throw_except=False)
  File "/home/lachlan/dev/pywren-ibm-cloud2/pywren_ibm_cloud/executor.py", line 439, in get_result
    result.append(f.result(throw_except=throw_except, internal_storage=self.internal_storage))
  File "/home/lachlan/dev/pywren-ibm-cloud2/pywren_ibm_cloud/future.py", line 224, in result
    raise FunctionException(self.executor_id, self.job_id, self.activation_id, self._exception)
TypeError: __init__() missing 1 required positional argument: 'exc_msg'
```

**Comment by JosepSampe** [comment]
@LachlanStuart Are you sure you updated the code? `FunctionException` is no longer used an completely deleted from the code.

**Comment by LachlanStuart** [comment]
@JosepSampe Ah, sorry, I forgot to change branch after checking out your fork. `throw_except` works as expected in #255. This issue can be resolved once that is merged.

**Comment by LachlanStuart** [comment]
Resolved in 1.4.1
